src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,145 @@
+import React, {
+  useCallback,
+  useRef,
+  useState,
+  useEffect,
+} from 'react';
+import cl from 'classnames';
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  todos: Todo[];
+  todo: Todo;
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>;
+}
+
+export const TodoItem: React.FC<Props> = React.memo(({
+  todos,
+  todo,
+  setTodos,
+}) => {
+  const [changedElement, setChangedElement] = useState('');
+  const [editId, setEditId] = useState<number | null>(null);
+
+  const [isEditing, setIsEditing] = useState(false);
+  const [focus, setFocus] = useState(false);
+
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  const handleTodoChange = () => {
+    let updatedTodos = [...todos];
+
+    updatedTodos = updatedTodos.map(elem => {
+      if (elem.id === editId) {
+        return { ...elem, title: changedElement };
+      }
+
+      return elem;
+    });
+
+    setTodos(updatedTodos);
+  };
+
+  const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    const keyDown = event.key;
+
+    if (keyDown === 'Enter') {
+      handleTodoChange();
+      setIsEditing(false);
+      setFocus(false);
+      setEditId(null);
+    }
+
+    if (keyDown === 'Escape') {
+      setEditId(null);
+      setIsEditing(false);
+      setFocus(false);
+    }
+  };
```

Comment:
you have duplicates here

it is make sense to create helper for those 3 actions